### PR TITLE
IIDA-119 Bug fixes for deferred revenue report

### PR DESCRIPTION
**Pull request by Edzelopez**

### CODE REVIEW

File: CRM/Financial/BAO/FinancialAccount.php [hunk 1]
```diff
@@ -439,16 +439,20 @@ public static function checkFinancialTypeHasDeferred($params, $contributionID =
...
-  public static function getAllDeferredFinancialAccount() {
-    $query = "SELECT cfa.id, cfa.name FROM civicrm_entity_financial_account ce
+  public static function getAllDeferredFinancialAccount($includeCode = FALSE) {
+    $query = "SELECT cfa.id, cfa.name, cfa.accounting_code FROM civicrm_entity_financial_account ce
 INNER JOIN civicrm_financial_account cfa ON ce.financial_account_id = cfa.id
 WHERE `entity_table` = 'civicrm_financial_type' AND cfa.is_active = 1 AND ce.account_relationship = %1 GROUP BY cfa.id";
     $deferredAccountRel = key(CRM_Core_PseudoConstant::accountOptionValues('account_relationship', NULL, " AND v.name LIKE 'Deferred Revenue Account is' "));
     $queryParams = array(1 => array($deferredAccountRel, 'Integer'));
     $dao = CRM_Core_DAO::executeQuery($query, $queryParams);
     $financialAccount = array();
```
Comment (pradpnayak) [inline CRM/Financial/BAO/FinancialAccount.php]: @Edzelopez  can you please rewrite this function to api?
    $result = civicrm_api3('EntityFinancialAccount', 'get', array(
      'sequential' => 1,
      'return' => array("financial_account_id.id", "financial_account_id.name", "financial_account_id.accounting_code"),
      'entity_table' => "civicrm_financial_type",
      'account_relationship' => "Deferred Revenue Account is",
    ));

I am worried if we submit this Eileen or Coleman won't accept this PR. :(

File: CRM/Financial/BAO/FinancialAccount.php [hunk 2]
```diff
@@ -439,16 +439,20 @@ public static function checkFinancialTypeHasDeferred($params, $contributionID =
    * @return array of Deferred Financial Account
    *
    */
-  public static function getAllDeferredFinancialAccount() {
-    $query = "SELECT cfa.id, cfa.name FROM civicrm_entity_financial_account ce
+  public static function getAllDeferredFinancialAccount($includeCode = FALSE) {
```
Comment (pradpnayak) [inline CRM/Financial/BAO/FinancialAccount.php]: grep -irn 'getAllDeferredFinancialAccount' . says it used at only one place i.e in Deferred Revenue report. Can we exclude this $includeCode

File: CRM/Financial/BAO/FinancialAccount.php [hunk 3]
```diff
@@ -439,16 +439,20 @@ public static function checkFinancialTypeHasDeferred($params, $contributionID =
...
     $deferredAccountRel = key(CRM_Core_PseudoConstant::accountOptionValues('account_relationship', NULL, " AND v.name LIKE 'Deferred Revenue Account is' "));
     $queryParams = array(1 => array($deferredAccountRel, 'Integer'));
     $dao = CRM_Core_DAO::executeQuery($query, $queryParams);
     $financialAccount = array();
     while ($dao->fetch()) {
-      $financialAccount[$dao->id] = $dao->name;
+      $label = $dao->name;
+      if ($includeCode) {
+        $label .= " ({$dao->accounting_code})" ;
+      }
```
Comment (pradpnayak) [inline CRM/Financial/BAO/FinancialAccount.php]: Same as above

File: CRM/Financial/BAO/FinancialAccount.php (hunk 1 above)
Comment (Edzelopez) [inline CRM/Financial/BAO/FinancialAccount.php]: Isn't this already submitted as a PR?

File: CRM/Financial/BAO/FinancialAccount.php (hunk 2 above)
Comment (Edzelopez) [inline CRM/Financial/BAO/FinancialAccount.php]: Sure

**Comment by Edzelopez** [comment]
@pradpnayak made the requested changes. Could you please check?

**Pull request merged**